morph_tool/sub_sectiontypes.py

```diff
@@ -0,0 +1,137 @@
+    """Return a dict of extended apical subtypes (trunk, oblique, tuft)."""
+    extended_types = {}
+
+    apical_section = neuron.sections[
+        apical_point_section_segment(neuron, tuft_percent=tuft_percent)[0]
+    ]
+    if apical_section is not None:
+        for section in iter_sections(neuron):
+            if section.type == SectionType.apical_dendrite:
+                extended_types[section.id] = "oblique"
+
+        for section in apical_section.ipreorder():
+            extended_types[section.id] = "tuft"
+
+        for section in apical_section.iupstream():
+            extended_types[section.id] = "trunk"
+    else:
+        for section in iter_sections(neuron):
+            if section.type == SectionType.apical_dendrite:
+                extended_types[section.id] = "apical"
+
+    for section in iter_sections(neuron):
+        if section.type == SectionType.basal_dendrite:
+            extended_types[section.id] = "basal"
+        if section.type == SectionType.axon:
+            extended_types[section.id] = "axon"
+
+    return extended_types
+
+
+def axon_subtype(neuron, axonal_section=None, direction=None, bbox=None, ignore_axis=2):
+    """Return a dict of extended axonal subtypes (main, collaterals)."""
+    extended_types = {}
+    if axonal_section is None:
+        axonal_section = neuron.sections[
+            axon_point_section(neuron, direction=direction, bbox=bbox, ignore_axis=ignore_axis)
+        ]
+        for section in iter_sections(neuron):
+            if section.type == SectionType.axon:
+                extended_types[section.id] = "collateral"
+
+        for section in axonal_section.iupstream():
+            extended_types[section.id] = "main"
+    else:
+
+        for section in iter_sections(neuron):
+            if section.type == SectionType.axon:
+                extended_types[section.id] = "axon"
+            if section.type == SectionType.basal_dendrite:
+                extended_types[section.id] = "basal"
+            if section.type == SectionType.apical_dendrite:
+                extended_types[section.id] = "apical"
+
+    return extended_types
+
+
+def set_apical_subtypes(neuron, tuft_percent=20):
+    """Set apical subtypes to a morphology.
+
+    WARNING: cannot save it to .asc file, use unset_subtype to revert
+    """
+    subtypes = apical_subtype(neuron, tuft_percent=tuft_percent)
+    sections = neuron.sections
+    for secid, subtype in subtypes.items():
+        sections[secid].morphio_section.type = APICAL_SUBTYPE_MAP[subtype]
+    return neuron
+
+
+def set_axon_subtypes(neuron, **kwargs):
+    """Set subtypes to a morphology.
+
+    WARNING: cannot save it to .asc file, use unset_subtype to revert
+    """
+    subtypes = axon_subtype(neuron, **kwargs)
+    sections = neuron.sections
+    for secid, subtype in subtypes.items():
+        sections[secid].morphio_section.type = AXON_SUBTYPE_MAP[subtype]
+    return neuron
+
+
+def _unset_subtypes(neuron, base_type):
```

Comment:
What about using a mapping of base types? So it is possible to use several custom types at the same time.

Reply:
yes, I should rather use the REV_ mapping here, so we can revert only specific custom types